lib/import/brca/providers/st_george/st_george_handler.rb

```diff
@@ -1,436 +1,354 @@
-              genotype_dup.add_status(9)
-              genotypes.append(genotype_dup)
-            end
-          end
+            return if record.raw_fields['moleculartestingtype'].blank?
 
-          def single_variant_full_screen(genotype, genotypes, positive_genes, record)
-            negative_gene = %w[BRCA1 BRCA2] - positive_genes
-            genotype_dup = genotype.dup
-            genotype_dup.add_gene(negative_gene.join)
-            genotype_dup.add_status(1)
-            genotypes.append(genotype_dup)
-            genotype.add_gene(positive_genes.join)
-            process_single_positive_variants(genotype, record)
-            process_single_protein(genotype, record)
-            genotypes.append(genotype)
-          end
+            return unless TEST_TYPE_MAP[record.raw_fields['moleculartestingtype']&.downcase&.strip]
 
-          def process_targeted_records(positive_genes, genotype, record, genotypes)
-            if normal?(record)
-              process_normal_targeted(genotype, record, genotypes)
-            elsif failed_test?(record)
-              process_failed_targeted(genotype, record, genotypes)
-            elsif positive_cdna?(record) || positive_exonvariant?(record)
-              process_positive_targeted(record, positive_genes, genotype, genotypes)
-            end
-            genotypes
+            genotype.add_molecular_testing_type(TEST_TYPE_MAP[record.raw_fields['moleculartestingtype']&.downcase&.strip])
           end
 
-          def process_normal_targeted(genotype, record, genotypes)
-            process_single_gene(genotype, record)
-            genotype.add_status(1)
-            genotypes.append(genotype)
-          end
+          def assign_test_scope(genotype, record)
+            # extract molecular testing type from the raw record
+            # map molecular testing type and assign to genotype using
+            # add_test_scope method from genotype_brca.rb
 
-          def process_failed_targeted(genotype, record, genotypes)
-            process_single_gene(genotype, record)
-            genotype.add_status(9)
-            genotypes.append(genotype)
-          end
+            testscope = record.raw_fields['moleculartestingtype']&.downcase&.strip
+            genotype.add_test_scope(TEST_SCOPE_MAP[testscope])
+            return genotype if genotype.attribute_map['genetictestscope'].present?
 
-          def process_positive_targeted(record, positive_genes, genotype, genotypes)
-            if record.raw_fields['genotype'].scan(CDNA_REGEX).size > 1
-              process_multiple_positive_variants(positive_genes, genotype, record, genotypes)
-            else
-              process_single_gene(genotype, record)
-              process_single_positive_variants(genotype, record)
-              process_single_protein(genotype, record)
-              genotypes.append(genotype)
-            end
-          end
+            genotype.add_test_scope(:no_genetictestscope)
```

Comment:
Unknown scope has been added here @lauramccluskey1 but not considered while processing the records ? Should you be having an else condition for these cases to be processed in fill_genotypes?

Reply:
Great spot @shilpigoeldev! I have just spoke to Fiona about this as its not clear what to do in this instance for st georges. She said that as there were no cases with no genetictestscope, this should give a big error instead. This means that if we see any cases in future new rules can be created. So I will add this as a logger message. @NImeson this applies to colorectal too

Reply:
Logger error added in [9104764](https://github.com/NHSDigital/data_management_system/pull/77/commits/9104764c8ae62c3c95b8e74cdc521840d4aabb83)